nengo/builder/operator.py

```diff
@@ -601,12 +601,19 @@ def make_step(self, signals, dt, rng):
         A = signals[self.A]
         Y = signals[self.Y]
 
-        def step_dotinc():
-            inc = A.dot(X)
-            if self.reshape:
-                inc = inc.reshape(Y.shape)
-            Y[...] += inc
-        return step_dotinc
+        if self.reshape:
+
+            def step_dotinc_reshape():
+                Y[...] += A.dot(X).reshape(Y.shape)
```

Comment:
I think it actually still makes a temporary variable for the reshaped matrix under the hood. In either case (old code or new code), that's just a matrix view, no data is copied.

Reply:
Yeah sorry I wasn't claiming that there's any change in anything other than the fact that the locals dictionary gets a new entry added to it, which is a very negligible but non-zero cost.